File: src/main/java/org/kite9/diagram/position/HPos.java
/**
 * 
 */
package org.kite9.diagram.position;

public enum HPos {
	LEFT, RIGHT;

	public Direction getDirection() {
		if (this.ordinal() == 0) {
			return Direction.LEFT;
		} else {
			return Direction.RIGHT;
		}
	}
	
	public static HPos getFromDirection(Direction d) {
		switch (d) {
		case LEFT:
			return LEFT;
		case RIGHT:
			return RIGHT;
		default:
			return null;
		}
	}
}

File: kite9-visualization/src/commonMain/kotlin/org/kite9/diagram/model/position/HPos.kt
/**
 *
 */
package org.kite9.diagram.model.position

import org.kite9.diagram.model.position.HPos

enum class HPos {
    LEFT, RIGHT;

    val direction: Direction
        get() = if (ordinal == 0) {
            Direction.LEFT
        } else {
            Direction.RIGHT
        }

    companion object {
        fun getFromDirection(d: Direction?): HPos? {
            return when (d) {
                Direction.LEFT -> LEFT
                Direction.RIGHT -> RIGHT
                else -> null
            }
        }
    }
}